### PR TITLE
WT-9384 Rename, split merge cppsuite files

### PR DESCRIPTION
**Pull request by etienneptl**
This PR is the continuity of https://github.com/wiredtiger/wiredtiger/pull/7981 that got reverted as the WiredTiger connection life cycle was not implemented correctly, see aa7d80d634d4b116196d5f768ff2f63bd723ab38.

### CODE REVIEW

**Review by tammybailey — APPROVED** [review]
Please run a patch build before merging! lgtm if patch build agrees.

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Pull request merged**
